spark-cassandra-connector/src/it/scala/com/datastax/spark/connector/rdd/CassandraJavaRDDSpec.scala

```diff
@@ -413,9 +417,10 @@ class CassandraJavaRDDSpec extends SparkCassandraITFlatSpecBase {
   }
 
   it should "allow to set limit" in {
-    val rdd = javaFunctions(sc).cassandraTable(ks, "test_table").limit(1L)
+    val limit = 1
+    val rdd = javaFunctions(sc).cassandraTable(ks, "limit_test_table").limit(limit.toLong)
     val result = rdd.collect()
-    result should have size 1
+    result.size shouldBe <= (rdd.getNumPartitions * limit)
```

Comment:
This is how `limit(...)` works. It limits the amount of RDD elements to `rdd.getNumPartitions * limit`.

Previously `getNumPartitions` was equal to `1`, so asserting ` result should have size 1` was fine.
Now, when there is a minimal threshold for partitions number, it gets a bit more complicated.
`val limit = 1` is as good as any I guess, should we change it some different value?